oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/K8sALSServiceMeshHTTPAnalysis.java

```diff
@@ -0,0 +1,275 @@
+                    .setDestServiceInstance(ingress.getServiceInstanceName())
+                    .setEndpoint(endpoint).setLatency((int)duration)
+                    .setResponseCode(Math.toIntExact(responseCode))
+                    .setStatus(status).setProtocol(protocol)
+                    .setDetectPoint(DetectPoint.server)
+                    .build();
+
+                logger.debug("Transformed ingress inbound mesh metric {}", metric);
+                forward(metric);
+
+                SocketAddress upstreamRemoteAddressSocketAddress = upstreamRemoteAddress.getSocketAddress();
+                ServiceMetaInfo targetService = find(upstreamRemoteAddressSocketAddress.getAddress(), upstreamRemoteAddressSocketAddress.getPortValue());
+
+                long outboundStartTime = startTime + formatAsLong(properties.getTimeToFirstUpstreamTxByte());
+                long outboundEndTime = startTime + formatAsLong(properties.getTimeToLastUpstreamRxByte());
+
+                ServiceMeshMetric outboundMetric = ServiceMeshMetric.newBuilder().setStartTime(outboundStartTime)
+                    .setEndTime(outboundEndTime)
+                    .setSourceServiceName(ingress.getServiceName())
+                    .setSourceServiceInstance(ingress.getServiceInstanceName())
+                    .setDestServiceName(targetService.getServiceName())
+                    .setDestServiceInstance(targetService.getServiceInstanceName())
+                    .setEndpoint(endpoint).setLatency((int)(outboundEndTime - outboundStartTime))
+                    .setResponseCode(Math.toIntExact(responseCode))
+                    .setStatus(status).setProtocol(protocol)
+                    .setDetectPoint(DetectPoint.client)
+                    .build();
+
+                logger.debug("Transformed ingress outbound mesh metric {}", outboundMetric);
+                forward(outboundMetric);
+            }
+        }
+    }
+
+    @Override public Role identify(StreamAccessLogsMessage.Identifier alsIdentifier,
+        Role prev) {
+        if (alsIdentifier != null) {
+            Node node = alsIdentifier.getNode();
+            if (node != null) {
+                String id = node.getId();
+                if (id.startsWith("router~")) {
+                    return Role.PROXY;
+                } else if (id.startsWith("sidecar~")) {
+                    return Role.SIDECAR;
+                }
+            }
+        }
+
+        return prev;
+    }
+
+    /**
+     * @param ip
+     * @param port
+     * @return found service info, or {@link ServiceMetaInfo#UNKNOWN} to represent not found.
+     */
+    protected ServiceMetaInfo find(String ip, int port) {
+        //TODO: go through API server to get target service info
+        // If can't get service or service instance name, set `UNKNOWN` string.
+        // Service instance name is pod name
+        // Service name should be deployment name
+        throw new UnsupportedOperationException("TODO");
```

Comment:
@hanahmily This should be implemented. Notice the `#return`, there is const for IP metadata not found